src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/LocalSCMListener.java

```diff
@@ -22,21 +21,20 @@
 import javax.annotation.CheckForNull;
 import javax.inject.Inject;
 import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
 
 @Extension
 public class LocalSCMListener extends SCMListener {
 
-    @Inject
-    private BuildStatusPoster buildStatusPoster;
+    private final List<BitbucketSCMCheckoutListener> checkoutListeners;
 
     public LocalSCMListener() {
+        checkoutListeners = Collections.emptyList();
     }
 
-    LocalSCMListener(BuildStatusPoster buildStatusPoster) {
-        this.buildStatusPoster = buildStatusPoster;
+    @Inject
+    LocalSCMListener(BuildStatusPoster buildStatusPoster, DeploymentPoster deploymentPoster) {
```

Comment:
I'm not sure of the best way to do this. Maybe I need to create a `Module` so that I can inject this as a list and do the wiring there? Thoughts?

Reply:
I think using modules (Guice MultiBinder) is the way to go.

```
class BitbucketScmCheckoutListenerModule extends AbstractModule {
    @Override
     protected void configure() {
        Multibinder.newSetBinder(BitbucketSCMCheckoutListener.class).addBinding().to(BuildStatusPoster.class):
        Multibinder.newSetBinder(BitbucketSCMCheckoutListener.class).addBinding().to(DeploymentPoster.class):
    }
}
```

then inject using

```
@Inject
    LocalSCMListener(Set<BitbucketSCMCheckoutListener> implementations) {
```
 [SO source](https://stackoverflow.com/questions/6085200/how-to-get-all-implementors-subclasses-of-an-interface-with-guice)